pkg/proxy/handlers/spacelister.go

```diff
@@ -37,26 +39,85 @@ func NewSpaceLister(app application.Application) *SpaceLister {
 
 func (s *SpaceLister) HandleSpaceListRequest(ctx echo.Context) error {
```

Comment:
I still think that having two separate handlers and use routes (https://github.com/codeready-toolchain/registration-service/blob/master/pkg/proxy/proxy.go#L127-L130) would make the code cleaner. This is why we have routes in the first place.

Reply:
I agree, and I can do that split. But as I briefly explained in the PR description this will introduce more changes (at least to the unit tests), and make the PR bigger. If this is not a concern then I'm fine with adding those changes to the current PR.

Reply:
It's totally fine to split your work in multiple PR which compliment each other. For example one PR which introduces an Informer. Another one which adds a new handler for Get. And one more for adding a role.
What I would avoid is splitting your work in the way that the following PR has to refactor the code you introduced in the previous PRs. Like adding everything in one handler and then another PR which refactors the handler and the code you just added. You see what I mean?

Reply:
yeah, that was a better way of splitting the PRs. My PR now has the new nstemplatetier informer and available roles, the refactor for adding the new handler for GET will be "minimal" in the handler code, it will be more unit tests separation from what I can see. 

My idea was to:
1) introduce the new available roles (current PR)
2) introduce the list of bindings (at this point there will be some code duplication probably the code in host operator for listing all spacebindings, see: https://github.com/codeready-toolchain/host-operator/blob/c68ff269240fd984839a19fb6743288f31381d1b/controllers/space/space_controller.go#L260-L313)

at this point we delivered the functionality to RHTAP, and we can focus on refactoring if needed:

3) refactor the spacebindings listing and extract common logic to toolchain-common in order use it in both host-operator and registration-service
4) refactor the existing List handler and create a Get handler

The main idea was to first deliver the functionality (and small improvements), and then making the existing code better once RHTAP is using it already.  
But of course I'm open to suggestions.

Reply:
Maybe I'm missing something but starting with two handlers instead of adding everything to the single one doesn't really make the implementation harder... But again, maybe I'm missing something :)

Reply:
No it doesn't make the implementation harder , it's just some more changes to review :). If this is not a concern, I'll add  the changes for the new handler in the current PR.